GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -383,7 +386,15 @@ public override bool TryRunInstaller(InstallActionWrapper installActionWrapper,
                             {
                                 if (!this.dryRun)
                                 {
-                                    this.RunInstaller(installerPath, processedArgs, out installerExitCode, out localError);
+                                    if (!string.IsNullOrEmpty(entry.Command))
+                                    {
+                                        this.RunInstaller(entry.Command, processedArgs, out installerExitCode, out localError);

```

Comment:
Who controls the command that is ran here?  Are there any security concerns?

Reply:
Should we scope this down or allow any command?

Reply:
The command comes from the "installer payload". Inside the "installer payload" is a manifest that describes the commands that should be run during upgrade / installation. 

> Are there any security concerns?

This is a good item to keep in mind - this is a quite critical component that we need to protect, and one that we have put a good deal of thought / effort into. If you have any suggestions on how we can minimize concerns here, I would love to follow up on them. We protect the components that control where we download packages from, we verify the integrity of the NuGet package that is downloaded (by default - you can currently opt-out of this verification, and net core does not currently support verification yet), and we download / unpack the package into locations that are only writable by root.

> Should we scope this down or allow any command

Interesting idea... It is quite broad and flexible right now (by design).

I will point out, that this change does not broaden the attack surface, as we already had the ability to run scripts in the upgrader package (by design).